epic: esl-animate (#698)
exadel-inc/esl: src/modules/esl-animate/README.md

# [ESL](https://exadel-inc.github.io/esl/) Animate

Version: _1.0.0-beta_.

Authors: *Anna-Mariia Petryk*, *Alexey Stsefanovich (ala'n)*, *Julia Murashko*.

**_Important Notice: the component is under beta version, it is tested and ready to use but be aware of its potential critical API changes._**

<a name="intro"></a>

ESL Animate is a module to animate items on viewport intersecting.

Features:
- Add class(es) when observed elements enter viewport area
- Support group animation that allows item delay its animation on the passed time after previously animated item
- Support forward and backward animations directions
- Support automatic re-animate after item exit viewport area
- JS API (Service) + Custom Element - Plugin for simple initialization

## ESLAnimateService

> ESLAnimateService - is a core of esl-animate module. Element is needs to be observed by ESLAnimateService 

Reviewer: ```suggestion
ESLAnimateService - is a core of esl-animate module. Element needs to be observed by ESLAnimateService 
```

in order to be animated.

### Service API:
- `ESLAnimateService.observe(els, config)` - method to start element observation
  - `els` - Element or array of Elements to observe and animate

>   - `config` - optional ESLAnimateConfig object to describe animate functionality behaviour

Reviewer: ```suggestion
  - `config` - an optional ESLAnimateConfig object to describe the behavior of the animation functionality
```

> - `ESLAnimateService.unobserve(els)` - method to unsubscribe ESLAnimateService from elements observation

Reviewer: ```suggestion
- `ESLAnimateService.unobserve(els)` - method to unsubscribe ESLAnimateService from observing elements
```
